res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -1,41 +1,126 @@
 var DDJ200 = {
-    jogDisabled: new Array(false, false, false, false, false)
+    fourDeckMode: false,
+    vDeckNo: [0, 1, 2],
+    vDeck1: {
+        syncEnabled: false, volMSB: 0, rateMSB: 0, jogDisabled: false
+    },
+    shiftPressed: {left: false, right: false},
+    jogCounter: 0
 };
 
-DDJ200.init = function() {};
+DDJ200.init = function() {
+    // creating associative arrays for the virtual decks
+    this.vDeck = {
+        1: this.vDeck1, 2: _.clone(this.vDeck1),
+        3: _.clone(this.vDeck1), 4: _.clone(this.vDeck1)
```

Comment:
I'm really not a fan of lodash. This is very unreadable IMHO.

Reply:
What alternative do you suggest?

Reply:
Something like this maybe?

```javascript
this.vDeck = {}
for (var i = 1; i <= 4; i++) {
    this.vDeck[i] = {
        syncEnabled: false,
        volMSB: 0,
        rateMSB: 0,
        jogDisabled: false,
    }
}
```